crates/bevy_wgpu/src/lib.rs

```diff
@@ -32,11 +32,39 @@ impl Plugin for WgpuPlugin {
 }
 
 pub fn wgpu_render_system(resources: &mut Resources) -> impl FnMut(&mut World, &mut Resources) {
-    let mut wgpu_renderer = pollster::block_on(WgpuRenderer::new());
+    let options = resources
+        .get_cloned::<WgpuOptions>()
+        .unwrap_or_else(|| WgpuOptions::new());
+    let mut wgpu_renderer = pollster::block_on(WgpuRenderer::new(options));
     let resource_context = WgpuRenderResourceContext::new(wgpu_renderer.device.clone());
     resources.insert::<Box<dyn RenderResourceContext>>(Box::new(resource_context.clone()));
     resources.insert(SharedBuffers::new(Box::new(resource_context)));
     move |world, resources| {
         wgpu_renderer.update(world, resources);
     }
 }
+
+#[derive(Clone)]
+pub struct WgpuOptions {
+    power_pref: WgpuPowerOptions,
+}
+
+#[derive(Clone)]
+pub enum WgpuPowerOptions {
+    HighPerformance,
+    Adaptive,
+    LowPower,
+}
+
+impl WgpuOptions {
+    pub fn new() -> WgpuOptions {
+        WgpuOptions {
+            power_pref: WgpuPowerOptions::HighPerformance,
+        }
+    }
+
+    pub fn power_preference(mut self, preference: WgpuPowerOptions) -> WgpuOptions {
+        self.power_pref = preference;
+        self
+    }
+}
```

Comment:
Implement a default.

```suggestion
}

impl Default for WgpuOptions {
    fn default() -> Self {
        Self::new()
    }
}
```